### PR TITLE
CB-10405 - Connectivity issue to Cordova VM

### PR DESCRIPTION
**Pull request by sarangan12**
Added Retry Logic to connect to cordova VM.

@dblotsky @riknoll @nikhilkh Please review and merge this PR?

### CODE REVIEW

File: lib/testcheck.js [hunk 1]
```diff
@@ -21,22 +21,40 @@
...
 
     var http = require("http"),
         url = require("url"),
         q = require("q");
 
+    function tryConnect(options, d, pendingnumberoftries, operation, successCallBack){
+        http.get(options, successCallBack)
+            .on("error", function(e){
+                if(pendingnumberoftries > 1){
+                    console.log("Connection attempt to " + operation + " failed. Will try after: " + delay + " milliseconds.");
```
Comment (nikhilkh) [inline lib/testcheck.js]: Should you be using the util log function here - as it has some nice functionality to put time etc?

Comment (nikhilkh) [inline lib/testcheck.js]: Also, consider logging e.message for every failure

**Comment by dblotsky** [comment]
Does this need to be fixed? It's an intermittent connectivity issue and already gets polled, so if some polls fail it's not a problem.

File: lib/testcheck.js (hunk 1 above)
Comment (sarangan12) [inline lib/testcheck.js]: @nikhilkh Logged the e.message and moved the log to use util.medicLog

**Comment by sarangan12** [comment]
@dblotsky Yes. We have seen the timeouts 4 times last week. I was able to connect to the VM (and get the same results) right at the same time it was failing in the CI. So, this is added as an additional step to ensure that if the system is not able to connect, it tries again after waiting for 15 seconds.

**Comment by dblotsky** [comment]
Oh, I think I'm mistaking this for `testwait`. `testcheck` does *not* poll and yes, this is necessary. My mistake! Do you think this can be factored out to be in `util` then, so that there is no duplication of retry logic with `testwait`? If it's too long a process, I guess we can have some duplication.

**Comment by sarangan12** [comment]
@dblotsky I think both of them use slightly different libs (request and http). So, for now, I am leaving it as is. But, I am marking this for further changes as we plan to move the slave machine itself to new one (soon :-))

**Comment by riknoll** [comment]
@dblotsky is this good to merge?

**Comment by dblotsky** [comment]
Alright, LGTM then.

**Pull request merged**